### PR TITLE
Fixes #5880 - Move test-simple-webapp to demos.

### LINKED ISSUE #5880: Move test-simple-webapp to demos
**Jetty version**
10.0.x

**Description**
There is a super-simple test webapp that would be handy to use in the documentation, and therefore it'll be better if it's a demo with a `[files]` section so that it can be used with `--add-module`.

### PR DESCRIPTION
**Pull request by sbordet**
Moved test-simple-webapp to demos.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

**Review by joakime — APPROVED** [review]
I'm good with it as is.
But it would be better if index.jsp was just renamed to index.html (as you don't have a JSP dependency declared in your demo-simple.mod

**Comment by janbartel** [comment]
@sbordet  if this is moved into demos, then as everything else in demos is deployed by the demos.mod file, then shouldn't this be too?  And shouldn't it be linked onto the root/index.html page as well?

**Review by janbartel — CHANGES_REQUESTED** [review]
See my comment on the PR.

**Comment by sbordet** [comment]
@janbartel I don't think it's worthwhile to add these 2 additional modules to the main demo, as we already have a "hello world" and there is no point in having 3 of them.

They can be enabled individually and they are mostly useful only for the documentation and for testing, so I think it's fine as is.

**Comment by janbartel** [comment]
@sbordet well, the hello worlds are actually different: one is a servlet and one is a static page. Also, your simplest webapp isn't the simplest as we don't require you to have a web.xml  if you just have static content :)

**Review by joakime — APPROVED** [review]
LGTM

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 1]
```diff
@@ -108,10 +108,10 @@ public void testQuickStartGenerationAndRun() throws Exception
 
         try (JettyHomeTester.Run run1 = distribution.start(args1))
         {
-            assertTrue(run1.awaitFor(5, TimeUnit.SECONDS));
+            assertTrue(run1.awaitFor(10, TimeUnit.SECONDS));
             assertEquals(0, run1.getExitValue());
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
Comment (janbartel) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: Should this be demo-simple-webapp or the demo-jsp-webapp? There's no demo-simple-jsp-webapp AFAIK.

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 2]
```diff
@@ -159,13 +159,13 @@ public void testSimpleWebAppWithJSP() throws Exception
...
         {
-            assertTrue(run1.awaitFor(5, TimeUnit.SECONDS));
+            assertTrue(run1.awaitFor(10, TimeUnit.SECONDS));
             assertEquals(0, run1.getExitValue());
 
             // Verify that --create-start-ini works
             assertTrue(Files.exists(jettyBase.resolve("start.ini")));
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
Comment (janbartel) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: Same as previous comment - think this should be either demo-jsp-webapp or demo-simple-webapp, not sure which.

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 3]
```diff
@@ -198,10 +198,10 @@ public void testSimpleWebAppWithJSPOnModulePath() throws Exception
         };
         try (JettyHomeTester.Run run1 = distribution.start(args1))
         {
-            assertTrue(run1.awaitFor(5, TimeUnit.SECONDS));
+            assertTrue(run1.awaitFor(10, TimeUnit.SECONDS));
             assertEquals(0, run1.getExitValue());
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
Comment (janbartel) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: See previous

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 4]
```diff
@@ -249,10 +249,10 @@ private void testSimpleWebAppWithJSPOverHTTP2(boolean ssl) throws Exception
...
```
Comment (janbartel) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: See previous

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 5]
```diff
@@ -308,7 +308,7 @@ public void testUnixSocket() throws Exception
             assertTrue(run1.awaitFor(30, TimeUnit.SECONDS));
             assertEquals(0, run1.getExitValue());
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
Comment (janbartel) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: See previous.

File: tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java [hunk 6]
```diff
@@ -345,11 +345,11 @@ public void testLog4j2ModuleWithSimpleWebAppWithJSP() throws Exception
         };
         try (JettyHomeTester.Run run1 = distribution.start(args1))
         {
-            assertTrue(run1.awaitFor(5, TimeUnit.SECONDS));
+            assertTrue(run1.awaitFor(10, TimeUnit.SECONDS));
             assertEquals(0, run1.getExitValue());
             assertTrue(Files.exists(jettyBase.resolve("resources/log4j2.xml")));
 
-            File war = distribution.resolveArtifact("org.eclipse.jetty.tests:test-simple-webapp:war:" + jettyVersion);
+            File war = distribution.resolveArtifact("org.eclipse.jetty.demos:demo-simple-jsp-webapp:war:" + jettyVersion);
```
Comment (janbartel) [inline tests/test-distribution/src/test/java/org/eclipse/jetty/tests/distribution/DistributionTests.java]: See previous.

**Review by janbartel — CHANGES_REQUESTED** [review]
The reference to the webapp is incorrect in several places in the DistributionTests - looks like a simple global search and replace error.

**Review by janbartel — APPROVED** [review]
LGTM

**Pull request merged**